core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -752,23 +751,20 @@ class DAGScheduler(
         submitStage(stage)
       }
     }
-    submitWaitingStages()
   }
 
   /**
    * Check for waiting stages which are now eligible for resubmission.
-   * Ordinarily run on every iteration of the event loop.
+   * Ordinarily run after the parent stage completed successfully.
    */
-  private def submitWaitingStages() {
-    // TODO: We might want to run this less often, when we are sure that something has become
-    // runnable that wasn't before.
+  private def submitWaitingChildStages(parent: Stage) {
     logTrace("Checking for newly runnable parent stages")
```

Comment:
Can you update this to say s"Checking if any dependencies of $parent are now runnable"